Marlin/src/lcd/ultralcd.cpp

```diff
@@ -233,6 +233,26 @@ millis_t next_button_update_ms;
 
 void MarlinUI::init() {
 
+  // MKS_MINI_12864 needs to do this early
+  #if ENABLED(MKS_MINI_12864)
+    #ifndef LCD_RESET_PIN
+      #define LCD_RESET_PIN LCD_PINS_RS
+    #endif
+    // Reset the MKS LCD
+    #if PIN_EXISTS(LCD_RESET)
+      OUT_WRITE(LCD_RESET_PIN, LOW);
+      _delay_ms(5);
+      OUT_WRITE(LCD_RESET_PIN, HIGH);
+      _delay_ms(5);
+    #endif
+    #if ENABLED(DEFAULT_LCD_CONTRAST)
+      u8g.setContrast(DEFAULT_LCD_CONTRAST);
+    #endif
```

Comment:
These bits should then go into the `init_lcd` function…

Reply:
btw, OUT_WRITE is a double function which do SET_OUTPUT + WRITE ... second one on same pin should use WRITE